engines/queen/metaengine.cpp

```diff
@@ -72,8 +78,68 @@ class QueenMetaEngine : public AdvancedMetaEngine<Queen::QueenGameDescription> {
 	int getMaximumSaveSlot() const override { return 99; }
 	bool removeSaveState(const char *target, int slot) const override;
 	int getAutosaveSlot() const override { return 99; }
+
+	void languageInput(Common::Keymap *perLanguage,const char *_commandKeys);
+	Common::KeymapArray initKeymaps(const char *target) const override;
+
 };
 
+const char *const _commandKeys[] = {
+	"ocmglptu", // English
+	"osbgpnre", // German
+	"ofdnepau", // French
+	"acsdgpqu", // Italian
+	"ocmglptu", // Hebrew
+	"acodmthu"  // Spanish
+};
+
+void QueenMetaEngine::languageInput(Common::Keymap *perLanguage, const char* _commandKeys) {
```

Comment:
should should rename the _commandKeys parameter to now shadow the global array name